src/Raven.Client/Documents/DocumentStore.cs

```diff
@@ -157,8 +157,7 @@ public override RequestExecutor GetRequestExecutor(string database = null)
         {
             AssertInitialized();
 
-            if (database == null)
-                database = Database;
+            database ??= Database ?? throw new ArgumentNullException(nameof(Database), "Document store database name cannot be empty");
```

Comment:
This needs to be done explicitly in BulkInsert. Like we do in Session.
Check here:

https://github.com/ravendb/ravendb/blob/52bb0beee6fe674b6f245433bbc82e442709e7d2/src/Raven.Client/Documents/Session/InMemoryDocumentSessionOperations.cs#L217-L219